src/protocolgame.cpp

```diff
@@ -3510,10 +3513,10 @@ void ProtocolGame::AddCreature(NetworkMessage& msg, const Creature* creature, bo
 	msg.addByte(player->canWalkthroughEx(creature) ? 0x00 : 0x01);
 }
 
-void ProtocolGame::AddCreatureIcons(NetworkMessage& msg, const Creature* creature)
+void ProtocolGame::AddCreatureIcons(NetworkMessage& msg, std::shared_ptr<const Creature> creature)
 {
 	const auto& creatureIcons = creature->getIcons();
-	if (const Monster* monster = creature->getMonster()) {
+	if (auto monster = creature->getMonster()) {
```

Comment:
Could this be `const auto`
```suggestion
	if (const auto monster = creature->getMonster()) {
```